rfc/ecs_e2e_tests.md

```diff
@@ -0,0 +1,371 @@
+
+Some things may impact this proposal which I'm hoping the RFC process can clear up:
+
+### About the test design
+
+* Whether any Jenkins/e2e test environment configuration would be impacted 
+  by this change but is not immediately obvious given the [`buildtool`](https://github.com/spinnaker/buildtool) code.
+* Whether testing a provider that relies on another provider (e.g, AWS) 
+  might cause issues with an unknown (to me) authentication or test mechanism.
+* What's the best way to test any new configuration + test code prior to merging? 
+* Unknown/show-stopping restrictions on the AWS account currently used during tests.
+* How many regions are running the e2e tests, and would we need a unique ECR repository for each?
+
+### About the future of testing / Spinnaker community CI tooling
+
+* Whether an alternative end to end testing mechanism (perhaps in the 
+  monorepo?) will be available soon enough to start building on in the next ~3 months.
+
+
+## Security, Privacy, and Compliance
+
+Creating new AWS resources may risk exposing parts of the test environment 
+via incorrect configuration. However there are established best practices 
+we can follow to mitigate this risk:
+
+* The ECR repository can be scoped to pull-only access by the IAM Role used for testing.
+* VPC subnets can be configured with security group(s) to prohibit 
+  ingress to only the specified load balancer.
+* (Optional) If we want to prevent ALL ingress/egress from the internet, ECS and ECR 
+  can also be accessed from within a completely private VPC (w/ additional 
+  configuration and cost considerations)
+* Additional ECS permissions needed on the test roles can be scoped 
+  down to specific resource types, names, and/or tags. For example: the IAM Role 
+  assumed by the test can be restricted to passing only roles that are assume-able by
+  Amazon ECS (vs. some other entity or service)
+
+If this proposal is accepted, more granular policies and configurations 
+will be provided prior to test contribution.
+
+
+## Operations
+
+The main drawback of this proposal is the additional overhead required 
+to set up, run, and debug potential test failures. However, since all 
+the tooling and dependencies needed to interact with AWS already exist 
+within the test stack, the additional surface area is mostly contained 
+to the contents of the test itself and any ECS-specific knowledge 
+needed to debug failures. 
+
+* Additional resources must be created and maintained in hosting account
+    * Mitigation: Use an infrastructure as code mechanism to modify/recreate (template can live alongside test, maybe?)
+* Test flakiness and overhead for build cops & release managers
+    * Mitigation: Pre-merge testing, maybe non-blocking test runs prior to triggering build fails? (needs input from community)
+* Upstream/AWS-side API errors (already a risk for AWS provider). 
+    * Mitigation: Hopefully rare, but can be clearly identified by API responses/error msgs
```

Comment:
We definitely see this from time to time, but as you say this is usually reasonably easy to identify based on error messages.